### DISCUSSION TITLE
The second law out of equilibrium

### DISCUSSION THREAD

**Discussion by josePereiro** [josePereiro/TankeFactory#6] [Discussions]
In an old discussion, I came across a statement [here](https://arxiv.org/abs/0808.0012) (section 3.4, page 59) that caught my attention: "Maxwell was the first to realize that the second law is not an absolute law (this was expressed in his popular textbook 'Theory of Heat' in 1871), that it 'has only statistical certainty,' and indeed, that in fluctuation phenomena 'the second law is continually being violated.' Such phenomena are not rare: just look out the window and you can see the sky is blue – a consequence of the scattering of light by density fluctuations in the atmosphere."

I still believe that the second law generally applies only to close-to-equilibrium states, or equilibrium-like states. That is, it is possible for an isolated system to have a transient where entropy decreases. However, as time approaches infinity and equilibrium is reached, the entropy will undoubtedly be at or very close to its maximum feasible value given the constraints of the system. 

What do you think?

**Comment by ta-rekt** [comment]
If you haven't seen it already, I highly recommend you this review: 
https://www.nature.com/articles/nphys3230

I don't think the second law applies only close to equilibrium. I think of the second law as saying "entropy production is always nonnegative". This statement is independent of being in equilibrium or not. Let me tell you why I think so.

Suppose you have a continuous-time Markov chain over a discrete set of states $I$. It has a rate matrix Q, and by doing $P(t) = e^{tQ}$ you get the probability transition matrix. Suppose you start the chain at some distribution $\lambda$. Then, at each time $t$, the probability distribution over states will be the vector $p(t) = \lambda P(t)$.

Now if you take the time-derivative of the Shannon entropy of this vector, you can do some simple tricks to show that it splits into two contributions, one of which is called entropy flow rate and is proportional to the heat exchanged with an equilibrium reservoir at fixed temperature, and the other is entropy production rate which is always nonnegative, and it doesn't care if you are at equilibrium in order to be nonnegative. It just says that if you do work on the system, some of the work will be dissipated and will show up as a positive EP rate.

![ctmc_ep_derivation](https://github.com/josePereiro/TankeFactory/assets/23706326/ef929bc9-4ac6-4557-a731-ad1263d843c0)
![ctmc_energetics](https://github.com/josePereiro/TankeFactory/assets/23706326/ad939e4e-80fd-4bea-a9b4-d390a73b08b9)

**Reply by josePereiro** [reply]
i) Yes, you shared this review previously. I made a fast scanned at the time but "me dió sartén" (which translate from cuban as "it hit me with a pan", which means "it was obscure to me"). I just add it to the Reading list and I'll will start to work on it. 

ii) I'll work in detail the CTMC example, but, I suppose that we must revise also the assumptions that are baked into the definition of the model dynamics, (eg. memoryless, the functional form of the transition rate, etc). How general is that at the time of modeling all conceivable systems? 
On the other hand, the above mentioned fluctuations argument is very strongly supporting that the tendency is not absolute, that you can have transients in the opposite direction. The strength of the 2nd law depends on the size of the system, being only exact at $N = \infty$. See Brownian motion, there you have indead 100% efficient heat-to-work transfers. The moment any 'dust' particle gain velocity, it reduces the temperature of the reservoir and so entropy decreases. Of course, this is a limiting case for a very small subsystem, but, the point is the same for smaller fluctuations in a larger one. This might get worse if you are outside equilibrium. I wouldn't be surprised that even for a common isolated system (eg. a gas in a cylinder) we can ingenier an initial microstate for where the evolution of the system toward equilibrium have a significant transient where entropy decreases. Such initial states will be extremely unlikely, but in theory feasible.

**Reply by ta-rekt** [reply]
Sorry, now I see what you mean. Yes I agree, the second law only applies probabilistically to small systems, so it can be violated by large enough fluctuations, and with small particles these fluctuations can be quite small as you said with Brownian motion. So as a "law" it only applies at $N = \infty$.

But, if you take expectation over all trajectories of a small particle, even if many of those trajectories will produce negative EP and therefore violate the second law, on average EP rate is always positive. That's what the derivation I showed you is trying to say. It's an expected value. This is why the stochastic thermodynamics people talk about the modern formulation of the second law to be: expected entropy production rate $\geq 0$. This statement is independent of being at equilibrium or not. 

Let's separate microscopic vs macroscopic, and equilibrium vs far from equilibrium. All four cases are possible. In macroscopic systems (equil. or not) fluctuations are negligible, so we don't worry about the second law being violated by them. Nevertheless, it still holds far from equilibrium.

There are two ways of being far from equilibrium: 1. you are on a transient, relaxing towards equilibrium but you are not there yet, or 2. you are being maintained far from equilibrium in a steady state. 

In the first case, the system could be absorbing heat from the environment and therefore reducing its total entropy, but its _internal_ entropy production will still be positive, and this is what the second law says.

In the second case, the system must be producing nonnegative entropy in order to remain far from equilibrium. If this were not the case, your system would be a heat engine with efficiency 1, which means it is reversible, which means it is operating infinitely slowly, which means it is very close to equilibrium.

Both these statements are also true, in expectation, for microscopic systems, even though they are continuously violated by fluctuations. But there is a deeper understanding of the second law which is only revealed by looking at microscopic systems, it's the fluctuation theorem 

$\log \frac{P_{forw}(\textbf{x})}{P_{rev}(\textbf{x})} = \Delta EP$

which says that accumulated entropy production is the log ratio of the forward and reverse probabilities of a fluctuating trajectory. The more likely the forward trajectory is relative to its time-reverse, the larger the entropy production will be, and therefore the more irreversible the system is. At equilibrium, both forward and reverse trajectories are indistinguishable and so EP = 0, and in general $EP \geq 0$. This applies even if trajectories begin and end away from equilibrium.

**Reply by josePereiro** [reply]
This is great. We can see equilibrium as a global attractor and the system to be performing and stochastic gradient descent (maximizing S). We are claiming that for must systems/models the landscape is such that a majority of realizations of the stochastic gradient will be overall efficient (positive average S production) regardless the initial point. An more amazing, this behaviour is independent to a lot of details about the systems, mostly depending on size (the thermodynamic limit). 

I found very interesting, I think related, the phenomenon of frustrations, where the system has a hard time reaching equilibrium.  I found [this](https://www.americanscientist.org/print/1984) article simple and motivating. Maybe we can discuss it too.
